spec/models/guest_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Guest, type: :model do
+  describe 'relationships' do
+    it{should have_many :guest_rooms}
+    it{should have_many :rooms}
+  end
+
+  describe "class methods" do
+    it "#count" do
+      guest1 = Guest.create!(name: "Bob", nights: 3)
+      guest2 = Guest.create!(name: "Shane", nights: 10)
+
+      expect(Guest.count_guests).to eq(2)
+
+      guest3 = Guest.create!(name: "Isaac", nights: 13)
+
+      expect(Guest.count_guests).to eq(3)
```

Comment:
The danger of using a class method in this scenario is that there are now 2 ways it can be used - called on a collection of ALL the guests, or called on a collection of the guests _associated to a Room_. The 2nd way is what the user story is asking for, so we should really be testing it in that way as well. As is, this doesn't fully cover the functionality of this method. 

What would make it fully covered would be to add some `Room` instances and relate some Guests to those Rooms already, and make sure the value returned is different for each Room instance. Example: 
```
room1 = Room.create!(...)
room2 = Room.create!(...)

GuestRoom.create(room: room1, guest: guest1)
GuestRoom.create(room: room1, guest: guest2)
GuestRoom.create(room: room2, guest: guest1)
```
Then our expect statements would include:
```
expect(room1.guests.count_guests).to eq(2)
expect(room2.guests.count_guests).to eq(1)
```

Reply:
Doing it this way ^ like I illustrated would cover the use case that the user story is asking for, and that's what I want to underline is important about writing model tests. You did cover this functionality in the feature test, but that's for intergrations - model tests are meant to be unit/bite-size/atomic tests that ensure functionality works in the way we intend it. Since you used it on an association and not on a Model in the application itself, we need to ensure that way is tested. Let me know if you want to talk about this more.